### PR TITLE
JITMs: update pre-connection messages for launch

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

This PR updates our pre-connection JITMs so they can be launched to all Jetpack sites. Until now, they were only available when setting a filter on your site

This PR:

﻿- removes that dependency to a filter to display the pre-connection messages.
- ~~updates the existing function used to display the ToS blurb, so that blurb can be used without echo'ing it right away.~~ Reverted that, we'll use custom ToS blurbs in those messages. See p1HpG7-bPy-p2#comment-46028
- updates the connection links in all pre-connection messages.

Screenshot 2021-04-23 at 18 09 40
Screenshot 2021-04-23 at 18 09 34
Screenshot 2021-04-23 at 18 09 24

~~A by-product of this change is that clicking on the set-up button in those messages launches the old Calypso flow instead of the in-place flow.~~ I fixed that in 33b944bc71ff085e45f83f0590bb9288afe1e32a

#### Jetpack product discussion

* Internal reference: p1HpG7-bPy-p2

#### Does this pull request change what data or activity we track or use?

* It does add a from parameter to JITM buttons in those pre-connection messages, but that from parameter isn't communicated to WordPress.com until you click to connect.

#### Testing instructions:

* On a brand new site, go to the Media screen, or Appearance > Widgets, or the Posts screen if you've published 5 posts or more.
* You should see JITMs inviting you to connect your site to WordPress.com.
* If you connect, you should notice the name of the message in the `from` parameter of the `wpcom_jpc_authorize_begin` Tracks event that is logged a few minutes later.

### CODE REVIEW

File: projects/plugins/jetpack/class-jetpack-pre-connection-jitms.php
Comment (leogermani) [inline projects/plugins/jetpack/class-jetpack-pre-connection-jitms.php:52]: I think it should be `'page' => 'jetpack#/setup'`

If we use it as it's used [here](https://github.com/Automattic/jetpack/blob/add%2Fsegmentation_params_to_registration/projects/plugins/jetpack/class.jetpack-connection-banner.php#L204) (plus the `from` param) it should work as expected.

The Set Up button on the big banner in the initial wp-admin dashboard is intercepted by a JS. But, in the end, it just [redirects the window to the URL I showed above](https://github.com/Automattic/jetpack/blob/add%2Fsegmentation_params_to_registration/projects/plugins/jetpack/_inc/connect-button.js#L59) (adding the `from` param). So I think it's fine is we just point it directly there:

**Comment by jeherve** [comment]
On Hold pending final wording on the messages. See p1HpG7-bPy-p2#comment-46045

**Comment by folletto** [comment]
Can we avoid the "Set up Jetpack" button to wrap?
Might be enough to use `white-space: nowrap;`.

**Comment by jeherve** [comment]
> On Hold pending final wording on the messages.

Done in 565c5f57672561aebdb93b3f3fd14221a41f05c0

**Comment by jeherve** [comment]
> Can we avoid the "Set up Jetpack" button to wrap?
> Might be enough to use `white-space: nowrap;`.

Sure thing! Done in ea8c71ee9aeb42d3bda8108239d04a9ca173702c

File: projects/plugins/jetpack/class-jetpack-pre-connection-jitms.php
Comment (jeherve) [reply projects/plugins/jetpack/class-jetpack-pre-connection-jitms.php]: 👍 this should take care of that:
53fb4b984bd800353da295e351ddcc0ab96912ff

**Comment by sdixon194** [comment]
This works for me! The only question I have, that I don't think is a blocker - is that this is supposed to go away even for a userless connection, right? If I go to connect, then leave before choosing the option to Approve, the message goes away, but the user connection isn't there.

I'm thinking that's expected since the benefits of the JITM don't require a user connection (unsure if there are plans to update/add JITM messages to prompt people to connect to their wpcom accounts).

**Review by sdixon194 — APPROVED** [review]

**Comment by jeherve** [comment]
> is that this is supposed to go away even for a userless connection, right? If I go to connect, then leave before choosing the option to Approve, the message goes away, but the user connection isn't there.

Yes, I think that's the expected behaviour with a user-less connection.

**Comment by jeherve** [comment]
r224749-wpcom

**Pull request merged**
